language/tools/move-cli/src/main.rs

```diff
@@ -31,26 +31,35 @@ struct Move {
     /// Directory storing Move resources, events, and module bytecodes produced by script execution.
     #[structopt(name = "move-data", long = "move-data", default_value = MOVE_DATA)]
     move_data: String,
-    /// Directory storing Move source files that will be compiled and loaded into the VM before
-    /// script execution.
-    #[structopt(name = "move-src", long = "move-src", default_value = MOVE_SRC)]
-    move_src: String,
     // Command to be run.
     #[structopt(subcommand)]
     cmd: Command,
 }
 
 #[derive(StructOpt)]
 enum Command {
-    /// Compile the modules in `move_data` and the given script (if any).
-    #[structopt(name = "compile")]
-    Compile {
-        /// Path to script to compile.
-        #[structopt(name = "script")]
-        script_file: Option<String>,
-        /// If true, commit the bytecodes for the compiled modules to disk.
-        #[structopt(long = "commit", short = "c")]
-        commit: bool,
+    /// Type check and verify the specified script and modules against the modules in `move_data`
+    #[structopt(name = "check")]
+    Check {
+        /// The source files to check
+        #[structopt(
+            name = "PATH_TO_SOURCE_FILE",
+            default_value = MOVE_SRC,
+        )]
+        source_files: Vec<String>,
+    },
+    #[structopt(name = "publish")]
+    Publish {
+        /// The source files containing modules to publish
+        #[structopt(
+            name = "PATH_TO_SOURCE_FILE",
+            default_value = MOVE_SRC,
+        )]
+        source_files: Vec<String>,
+        /// If set, the effects of executing `script_file` (i.e., published, updated, and
+        /// deleted resources) will NOT be committed to disk.
+        #[structopt(long = "dry-run")]
+        dry_run: bool,
```

Comment:
I like the idea of making commits the default! Wondering about `dry-run` name though... would be nice to have something with a logical `short` (`-d` isn't great because it typically means something else). What about `read-only`?

Reply:
I like dry-run as I've personally seen it more often with the command line tools I've interacted with. 

Thoughts on `-n` for the short? It seems to be what git and ninja use